rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -311,9 +311,26 @@ func (s *RollupSyncService) getChunkRanges(batchIndex uint64, vLog *types.Log) (
 		return []*rawdb.ChunkBlockRange{{StartBlockNumber: 0, EndBlockNumber: 0}}, nil
 	}
 
-	tx, _, err := s.client.client.TransactionByHash(context.Background(), vLog.TxHash)
+	tx, _, err := s.client.client.TransactionByHash(s.ctx, vLog.TxHash)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get transaction, err: %w", err)
+		log.Warn("failed to get transaction by hash, probably an unindexing transaction, fetching the whole block to get the transaction",
```

Comment:
Better to add a `Debug` log, otherwise if this is triggered the node's logs will be flooded with this message

Reply:
make sense, changed in [88c5e22](https://github.com/scroll-tech/go-ethereum/pull/601/commits/88c5e2217a4757cc49e812592814712c811de5ca).